litex/soc/cores/spi_xip/core.py

```diff
@@ -0,0 +1,79 @@
+from migen import *
+from migen.genlib.fsm import FSM, NextState
+
+from litex.soc.cores.spi_xip.common import *
+from litex.soc.interconnect import wishbone
+
+class LiteSPICore(Module):
+    def __init__(self, phy, endianness="big"):
+        self.bus = bus = wishbone.Interface()
+        self.submodules.fsm = fsm = FSM(reset_state="IDLE")
+        bus_read = Signal()
+
+        self.comb += bus_read.eq(bus.cyc & bus.stb & ~bus.we)
+
+        if endianness == "big":
+            self.comb += bus.dat_r.eq(phy.source.data)
+        else:
+            for i in range(4):
+                self.comb += bus.dat_r[i*8:(i+1)*8].eq(phy.source.data[(3-i)*8:(4-i)*8])
+
```

Comment:
There is a` reverse_bytes` function in `litex.gen.common` that you could probably use here:
````
self.comb += bus.dat_r.eq(phy.source.data if endianness == "big" else reverse_bytes(phy.source.data))
````

Reply:
Thanks, I was not aware of that function